src/net_message.h

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NET_MESSAGE_H
+#define BITCOIN_NET_MESSAGE_H
+
+#include <hash.h>
+#include <protocol.h>
+#include <streams.h>
+#include <uint256.h>
+
+/** Maximum length of incoming protocol messages (no message over 4 MB is currently acceptable). */
+static const unsigned int MAX_PROTOCOL_MESSAGE_LENGTH = 4 * 1000 * 1000;
+
+// base class for format agnostic network messages
+class NetMessageBase
+{
+public:
+    CDataStream vRecv; // received message data
+    int64_t nTime;     // time (in microseconds) of message receipt.
```

Comment:
nit, `int64_t nTime{0};` and remove from the constructor.

Reply:
That's a nice nit @promag. Alternatively, `nTime` can be initialized with an empty initializer list `int64_t nTime {};`

Reply:
Agree. Though I'd say scope of the PR is not to refactor code parts that where moved 1:1. Lets do further refactors in another PR.

Reply:
SGTM.